Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.cpp

```diff
@@ -274,7 +276,8 @@ namespace ROS2::Utils
         urdfFileCrc.Add(urdfFilename);
         const AZ::IO::Path urdfPath(urdfFilename);
         const AZStd::string directoryNameTmp = AZStd::string::format("$tmp_%u.tmp", AZ::u32(urdfFileCrc));
-        const AZStd::string directoryNameDst = AZStd::string::format("%u_%s", AZ::u32(urdfFileCrc), urdfPath.Stem().String().c_str());
+        const AZStd::string directoryNameDst =
+            AZStd::string::format("%u_%s%s", AZ::u32(urdfFileCrc), urdfPath.Stem().String().c_str(), outputDirSuffix.c_str());
```

Comment:
This format can be updated to format the suffix from the string view.
Also using the string type of "AZ::IO::FixedMaxPathString" which is a type alias of  `AZStd::fixed_string<1024>` avoids a heap allocation here.
Information about fixed_string is documented in its [header file](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/std/string/fixed_string.h#L20-L25)
```suggestion
        const auto directoryNameDst =
            AZ::IO::FixedMaxPathString::format("%u_%.*s%.*s", AZ::u32(urdfFileCrc), AZ_PATH_ARG(urdfPath.Stem()), AZ_STRING_ARG(outputDirSuffix));
```